File: C#/Hashmap & Set/1207. Unique Number of Occurrences.cs
/* 1207. Unique Number of Occurrences
Difficulty: Easy
Submission Link: https://leetcode.com/problems/unique-number-of-occurrences/submissions/1148973580/ */

public class Solution {
    public bool UniqueOccurrences(int[] arr) {
        Dictionary<int,int> frequency = new Dictionary<int, int>();

        foreach(int num in arr)
            frequency[num] = frequency.ContainsKey(num) ? frequency[num]+1 : 1; 
        HashSet<int> storage = new HashSet<int>();
        foreach(var key in frequency)
            if(storage.Contains(key.Value)) return false;
            else storage.Add(key.Value);
        return true;
    }
}

File: JavaScript/Hashmap & Set/1207. Unique Number of Occurrences.js
/* 1207. Unique Number of Occurrences
Difficulty: Easy
Submission Link: https://leetcode.com/problems/unique-number-of-occurrences/submissions/1148977072/ */

var uniqueOccurrences = function(arr) {
    const frequency = new Map();
    for(let num of arr)
        if(frequency.has(num)) frequency.set(num, frequency.get(num)+1);
        else frequency.set(num,1);
    const storage = new Set(Array.from(frequency.values()));
    return frequency.size == storage.size;
};